zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/RoleState.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.immutable;
+
+public interface RoleState {
+
+  enum EntityType {
+    USER((byte) 0);
+
+    byte value;
+
+    EntityType(final byte value) {
+      this.value = value;
+    }
```

Comment:
🔧 Could extract this out of the `RoleState`? The `EntityType` can be reused for other state classes later on (TenantState, GroupState)